### PR TITLE
expression: implement vectorized evaluation for `builtinLastInsertIDSig`

### PR DESCRIPTION
**Pull request by tangwz**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinLastInsertIDSig.
Issue: #12103

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinInfoFunc/builtinLastInsertIDSig-VecBuiltinFunc-4               1818188               656 ns/op               0 B/op            0 allocs/op
BenchmarkVectorizedBuiltinInfoFunc/builtinLastInsertIDSig-NonVecBuiltinFunc-4              84112             13791 ns/op               0 B/op            0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_info_vec_test.go [hunk 1]
```diff
@@ -14,6 +14,7 @@
 package expression
 
 import (
+	"github.com/pingcap/tidb/types"
```
Comment (Reminiscent) [inline expression/builtin_info_vec_test.go]: Please format the header.

File: expression/builtin_info_vec_test.go [hunk 2]
```diff
@@ -31,6 +32,9 @@ var vecBuiltinInfoCases = map[string][]vecExprBenchCase{
 	ast.CurrentRole:    {},
 	ast.TiDBIsDDLOwner: {},
 	ast.ConnectionID:   {},
+	ast.LastInsertId: {
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{}},
```
Comment (Reminiscent) [inline expression/builtin_info_vec_test.go]: Should be ```{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETInt}},```

Comment (tangwz) [reply expression/builtin_info_vec_test.go]: why? I think `LAST_INSERT_ID()` has no args/If add this it must be fuction `builtinLastInsertIDWithIDSig`.

Comment (tangwz) [reply expression/builtin_info_vec_test.go]: @Reminiscent PTAL

**Comment by tangwz** [comment]
@Reminiscent PTAL

**Pull request merged**